src/main/java/net/dries007/tfc/world/layer/TFCLayers.java

```diff
@@ -156,7 +155,7 @@ public static AreaFactory createOverworldRockLayer(RegionGenerator generator, lo
         AreaFactory layer;
 
         layer = RegionRockLayer.INSTANCE.apply(regionLayer); // Grid scale (128x)
-        for (int i = 0; i < Units.GRID_BITS - 1; i++)
+        for (int i = 0; i < generator.getRockLayerScale(); i++)
```

Comment:
This is wrong. The region is still generating at grid scale and these zoom layers are necessary for matching that scale. This will offset everything and not just make it smaller but make the layer type completely offset from the rock.

Reply:
I guess it's finally time to do image generation of rock layers along with biomes in RegionGeneratorTest so I can visualize what you mean.